### PR TITLE
Change default local parallelism from 2 to 4 for avro/file/json sources

### LINKED ISSUE #2352: Increase parallelism of avro source
Avro source seems quite CPU intensive when reading and can benefit from higher default parallelism. For example, going from 2 to 8 parallelism cut the reading speed in half.

### PR DESCRIPTION
**Pull request by gurbuzali**
Change default local parallelism from 2 to 4 for avro/file/json sources

Checklist
- [X] Tags Set
- [X] Milestone Set
- [X] Any breaking changes are documented
- [NA] New public APIs have `@Nonnull/@Nullable` annotations
- [NA] New public APIs have `@since` tags in Javadoc
- [NA] For code samples, code sample main readme is updated

Links to issues fixed (if any):

Fixes #2352

List of breaking changes:

* Change default preferedLocalParallelism from 2 to 4

### CODE REVIEW

File: extensions/avro/src/main/java/com/hazelcast/jet/avro/AvroSourceBuilder.java [hunk 1]
```diff
@@ -82,6 +82,18 @@
...
     }
 
+    /**
+     * Sets the preferred local parallelism of the source. Default values is
+     * {@code 4}.
+     *
+     * @since 4.2
+     */
+    @Nonnull
+    public AvroSourceBuilder<D> preferredLocalParallelism(int preferredLocalParallelism) {
```
Comment (cangencer) [inline extensions/avro/src/main/java/com/hazelcast/jet/avro/AvroSourceBuilder.java:92]: do we need this option? I think just to change the default would have been enough. Do we have this in any other source?

Comment (gurbuzali) [reply extensions/avro/src/main/java/com/hazelcast/jet/avro/AvroSourceBuilder.java:92]: I don't think we have, I'll remove then. So wdyt about `4` as default ?
Should we set it only for Avro source ? or both (avro and file) ?

**Comment by cangencer** [comment]
We can use for 4 for both, as json parsing can also benefit from it

**Comment by viliam-durina** [comment]
We normally document the default local parallelism in javadoc, most methods in `Sources` do, those for files don't.

**Comment by gurbuzali** [comment]
Yes, we document them on builders, will change accordingly.

**Pull request merged**